### PR TITLE
feat(profiles): Accept profile chunks from Android

### PR DESCRIPTION
**Pull request by phacops**
This PR is adding support for profile chunks coming from the Android SDK.

### CODE REVIEW

File: relay-profiling/src/android/chunk.rs [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+fn parse_chunk(payload: &[u8]) -> Result<Chunk, ProfileError> {
+    let d = &mut serde_json::Deserializer::from_slice(payload);
+    let mut profile: Chunk = match serde_path_to_error::deserialize(d) {
+        Ok(p) => p,
+        Err(e) => {
+            println!("{:#?}", e);
+            return Err(ProfileError::InvalidJson(e));
+        }
+    };
+    // .map_err(ProfileError::InvalidJson)?;
```
Comment (iambriccardo) [inline relay-profiling/src/android/chunk.rs]:
```suggestion
```

File: relay-profiling/src/android/chunk.rs [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    }
+
+    // Use duration given by the profiler and not reported by the SDK.
+    profile.metadata.duration_ns = profile.profile.elapsed_time.as_nanos() as u64;
+    profile.metadata.timestamp = profile
+        .profile
+        .start_time
+        .timestamp_nanos_opt()
+        .unwrap_or_default() as f64
+        * 1e-9;
```
Comment (iambriccardo) [inline relay-profiling/src/android/chunk.rs]: nit: you could avoid the manual conversion.

let start_time_duration = Duration::from_nanos(
    profile.profile.start_time.timestamp_nanos_opt().unwrap_or_default() as u64,
);
profile.metadata.timestamp = start_time_duration.as_secs_f64();

File: relay-profiling/src/types.rs [hunk 3]
```diff
@@ -0,0 +1,14 @@
+use serde::{Deserialize, Serialize};
+
+use crate::native_debug_image::NativeDebugImage;
+
+#[derive(Default, Debug, Serialize, Deserialize, Clone)]
+pub struct DebugMeta {
```
Comment (iambriccardo) [inline relay-profiling/src/types.rs:7]: Can you document these types?

**Review by iambriccardo — APPROVED** [review]
LGTM, left some comments

File: relay-profiling/src/android/chunk.rs (hunk 1 above)
Comment (phacops) [reply relay-profiling/src/android/chunk.rs]: This was left over from debug, removed.

File: relay-profiling/src/types.rs (hunk 3 above)
Comment (phacops) [reply relay-profiling/src/types.rs:7]: These are the same fields as in Relay but serde-friendly. I added a comment to explain that.

**Review by viglia — APPROVED** [review]

File: relay-profiling/src/utils.rs [hunk 4]
```diff
@@ -47,3 +49,23 @@ where
...
+        "python" => "sentry.python",
+        "ruby" => "sentry.ruby",
+        "rust" => "sentry.rust",
+        _ => "unknown",
+    };
+    ClientSdk {
+        name: sdk_name.into(),
+        version: "unknown".into(),
+    }
+}
```
Comment (Zylphrex) [inline relay-profiling/src/utils.rs]: Wouldnt it be better to return `Option<ClientSdk>` and use `None` for the `unknown` cases instead?

**Review by Zylphrex — APPROVED** [review]
LGTM

**Pull request merged**